processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -748,7 +808,7 @@ public int size()
             channel.write(ByteBuffer.wrap(specBytes));
             serdeficator.write(channel);
             channel.close();
-          } else if (filename.startsWith("met_")) {
+          } else if (filename.startsWith("met_") || filename.startsWith("numeric_dim_")) {
```

Comment:
By this methodology, is every dimension type going to have a condition here?

Dimension name assumptions need revamped overall, see https://github.com/druid-io/druid/pull/3118 for related code.

I'm not convinced such a change would be in scope for this PR, but if you have any thoughts on ways to handle new dimension types in the future it would be nice to capture them for posterity.